src/sql/src/names.rs

```diff
@@ -1176,6 +1178,13 @@ pub enum CommentObjectId {
     ClusterReplica((ClusterId, ReplicaId)),
 }
 
+#[derive(Debug, Clone, Copy)]
+struct ItemResolutionConfig {
```

Comment:
Once #23789 is fully implemented, this should be able to get reduced to an enum:

```rust
enum ItemResolutionConfig {
  Functions,
  Types,
  Relations
}
```

I think this might be worth noting given that kind of inordinate complexity this spools into.

If this makes sense, glad to push the comment up but want to make sure that checks out w/ you before pushing to your branch.

Reply:
It _largely_ checks out for me, though I'm not sure that #23789 will be a strict simplification. The `DOC ON` stuff gets potentially more complicated, because I think you'll be able to leave comments on _either_ the type column or the relation column, rather than just on the relation column. I think? I'm not exactly sure how distinct the table and the type are in PostgreSQL.